src/components/sequencing/form/NumEditor.svelte

```diff
@@ -7,59 +7,37 @@
   // (("e" | "E") ("+" | "-")? ("_" | @digit)+)? |
   // @digit ("_" | @digit)* "n" |
 
-  import { isFswCommandArgumentUnsigned, type NumberArg } from '../../../utilities/codemirror/codemirror-utils';
-
-  // const PAT_INT = "^[-+]?\\d+$";
-  // const PAT_FLOAT = "^[-+]?\\d+\.?\\d*$";
+  import { isFswCommandArgumentUnsigned, type NumberArg } from './../../../utilities/codemirror/codemirror-utils';
 
   export let argDef: NumberArg;
-  export let initVal: string;
-  export let setInEditor: (val: string) => void;
+  export let initVal: number;
+  export let setInEditor: (val: number) => void;
 
   let max: number = Infinity;
   let min: number = -Infinity;
-  let value: string;
-
-  // $: pattern = isFswCommandArgumentUnsigned(argDef) ? PAT_INT : PAT_FLOAT;
+  let value: number;
 
   $: max = argDef.range?.max ?? Infinity;
   $: min = argDef.range?.min ?? (isFswCommandArgumentUnsigned(argDef) ? 0 : -Infinity);
   $: value = initVal;
   $: valFloat = Number(value);
   $: {
-    if (value && !isNaN(valFloat)) {
+    if (typeof value === 'number' && !isNaN(valFloat)) {
       setInEditor(value);
     }
   }
-
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  function customValidate(e: Event & { currentTarget: EventTarget & HTMLInputElement }) {
-    // const val = (e.target as HTMLInputElement).value;
-    const numVal = Number(value);
-    if (numVal < min) {
-      (e.target as HTMLInputElement).setCustomValidity('value is too low');
-    } else if (numVal > max) {
-      (e.target as HTMLInputElement).setCustomValidity('value is too low');
-    } else {
-      (e.target as HTMLInputElement).setCustomValidity('');
-    }
-  }
 </script>
 
 <div>
-  <input class="st-input" type="string" bind:value required />
-  <!-- {pattern} -->
-  <!-- on:input={customValidate} -->
-  {#if typeof min === 'number' && typeof max === 'number' && min === max && valFloat !== max}
-    <button on:click={() => setInEditor(max.toString())} title="Set to allowed value">{max}</button>
+  <input class="st-input w-100" type="number" bind:value required {min} {max} step="any" />
```

Comment:
Interesting. I needed to use `any` for `step` so that html input validity worked for floats (without a lot of fussing, see https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/step for the specifics) but it looks like setting `step` to `any` means that the browser still uses a step of 1 by default. Looks like the browser won't do less than a step to hit the max, oddly. We could maybe capture the specific up and down arrow behaviors on these inputs and perform the stepping ourselves though this feels like an edge case that may not be worth solving given that you can just input the number yourself?

Reply:
Ah so it's a browser specific quirk? If so then I agree, it's probably not worth trying to fix the browser behavior.

Reply:
Yeah I think this is just how the native behavior works, imagine it'd be the same across all browsers. Figure if it ends up being an issue we could put more time into a better solution